File: android-java/JavaMaps/app/src/main/java/com/tnqr/javamaps/adapter/PlaceAdapter.java
package com.tnqr.javamaps.adapter;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.constraintlayout.widget.Placeholder;
import androidx.recyclerview.widget.RecyclerView;

import com.tnqr.javamaps.databinding.RecyclerRowBinding;
import com.tnqr.javamaps.model.Place;
import com.tnqr.javamaps.view.MapsActivity;

import java.util.List;

public class PlaceAdapter extends RecyclerView.Adapter<PlaceAdapter.PlaceHolder> {

    //adaptere gelecek listeyi alıp bu listeyi kullanmak için oluşturduğumuz değişken ve consructer
    List<Place> places;

    public PlaceAdapter(List<Place> places) {
        this.places = places;
    }

    @NonNull
    @Override

    //
    public PlaceHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        RecyclerRowBinding recyclerRowBinding = RecyclerRowBinding.inflate(LayoutInflater.from(parent.getContext()),parent,false);

        return new PlaceHolder(recyclerRowBinding);
    }

    @Override
    //bağlanan görünmündeki her elemanın posisyonunu göre işlemler yapabilceğimiz bölüm
    public void onBindViewHolder(@NonNull PlaceHolder holder, int position) {
        holder.recyclerRowBinding.recylerViewTextView.setText(places.get(position).name);

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent =new Intent(holder.itemView.getContext(), MapsActivity.class);
                holder.itemView.getContext().startActivity(intent);
            }
        });


    }


    //listedeki eleman sayısını tutacak kısım
    @Override
    public int getItemCount() {
        return places.size();
    }


//satır olarak olan görünümü gelip bağlama işlemi
    public class PlaceHolder extends RecyclerView.ViewHolder{

       RecyclerRowBinding recyclerRowBinding;
        public PlaceHolder(RecyclerRowBinding recyclerRowBinding) {
            super(recyclerRowBinding.getRoot());
            this.recyclerRowBinding=recyclerRowBinding;
        }
    }
}

File: android-kotlin/GoogleMaps/app/src/main/java/com/example/googlemaps/adapter/PlaceAdapter.kt
package com.example.googlemaps.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.googlemaps.BitmapByteArrayConverter
import com.example.googlemaps.R
import com.example.googlemaps.database.Place

class PlaceAdapter (private val context: Context, private val placeList : List<Place>)
    :RecyclerView.Adapter<PlaceAdapter.PlaceCardHolder>(){


    class PlaceCardHolder(itemView : View) : RecyclerView.ViewHolder(itemView){
        val cardImage = itemView.findViewById<ImageView>(R.id.card_image)
        val placeName = itemView.findViewById<TextView>(R.id.card_placeTxt)
        val placeComment = itemView.findViewById<TextView>(R.id.card_commentTxt)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PlaceCardHolder {
       val view = LayoutInflater.from(parent.context).inflate(R.layout.place_card_layout,parent,false)
        return PlaceCardHolder(view)
    }

    override fun onBindViewHolder(holder: PlaceCardHolder, position: Int) {
        holder.placeName.text = placeList[position].name.toString()
        holder.placeComment.text = placeList[position].comment.toString()

        val currentByteArray = placeList[position].image
        val currentBitmap = BitmapByteArrayConverter.byteArrayToBitmap(currentByteArray)
        holder.cardImage.setImageBitmap(currentBitmap)
    }

    override fun getItemCount(): Int {
        return placeList.size
    }

}